program/src/instruction.rs

```diff
@@ -151,18 +151,26 @@ accounts_struct! {
         },
         pub user {
             is_signer: true,
+            // Is writable due to transfer (system_instruction::transfer) from user to
+            // reserve_account
             is_writable: true,
         },
         pub recipient {
             is_signer: false,
+            // Is writable due to mint to (spl_token::instruction::mint_to) recipient from
+            // st_sol_mint
             is_writable: true,
         },
         pub st_sol_mint {
             is_signer: false,
+            // Is writable due to mint to (spl_token::instruction::mint_to) recipient from
```

Comment:
It is writable because of the `mint_to`, but not due to the recipient, it’s because the mint stores the total supply.

Reply:
![image](https://user-images.githubusercontent.com/57174310/129000391-c5ada214-bbc0-49f0-b564-452ba8a0e252.png)

But the non-read-only accounts In mint_to (above) are the st_sol_mint and the recipient.